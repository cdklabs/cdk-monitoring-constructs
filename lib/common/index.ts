export * from "./alarm";
export * from "./metric";
export * from "./monitoring";
export * from "./url";
export * from "./widget";
export * from "./strings";
