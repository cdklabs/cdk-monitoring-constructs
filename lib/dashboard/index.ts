export * from "./BitmapDashboard";
export * from "./DashboardRenderingPreference";
export * from "./DynamicDashboardSegment";
export * from "./DashboardSegment";
export * from "./DashboardWithBitmapCopy";
export * from "./DefaultDashboardFactory";
export * from "./DynamicDashboardFactory";
export * from "./DefaultWidgetFactory";
export * from "./IDashboardFactory";
export * from "./IDynamicDashboardFactory";
export * from "./IWidgetFactory";
export * from "./MonitoringNamingStrategy";
export * from "./SingleWidgetDashboardSegment";
export * from "./widget";
