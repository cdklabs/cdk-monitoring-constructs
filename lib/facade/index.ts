export * from "./IMonitoringAspect";
export * from "./MonitoringFacade";
export * from "./MonitoringAspect";
