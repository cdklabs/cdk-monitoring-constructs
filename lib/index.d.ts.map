{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAC;AACzB,cAAc,aAAa,CAAC;AAC5B,cAAc,UAAU,CAAC;AACzB,cAAc,cAAc,CAAC"}