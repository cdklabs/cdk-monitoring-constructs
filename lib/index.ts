export * from "./common";
export * from "./dashboard";
export * from "./facade";
export * from "./monitoring";
