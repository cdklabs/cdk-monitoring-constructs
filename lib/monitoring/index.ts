export * from "./aws-acm";
export * from "./aws-apigateway";
export * from "./aws-apigatewayv2";
export * from "./aws-appsync";
export * from "./aws-billing";
export * from "./aws-cloudfront";
export * from "./aws-cloudwatch";
export * from "./aws-codebuild";
export * from "./aws-docdb";
export * from "./aws-dynamo";
export * from "./aws-ec2";
export * from "./aws-ecs-patterns";
export * from "./aws-elasticache";
export * from "./aws-glue";
export * from "./aws-kinesis";
export * from "./aws-kinesisanalytics";
export * from "./aws-lambda";
export * from "./aws-loadbalancing";
export * from "./aws-opensearch";
export * from "./aws-rds";
export * from "./aws-redshift";
export * from "./aws-s3";
export * from "./aws-secretsmanager";
export * from "./aws-sns";
export * from "./aws-sqs";
export * from "./aws-step-functions";
export * from "./aws-synthetics";
export * from "./aws-wafv2";
export * from "./custom";
export * from "./fluentbit";
