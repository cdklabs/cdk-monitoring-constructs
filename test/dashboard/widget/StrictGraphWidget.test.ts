import {
  HorizontalAnnotation,
  Metric,
  VerticalAnnotation,
  VerticalShading,
  YAxisProps,
} from "aws-cdk-lib/aws-cloudwatch";
import { DoubleAxisGraphWidget, SingleAxisGraphWidget } from "../../../lib";

const DummyMetric1 = new Metric({ namespace: "Dummy", metricName: "Metric1" });
const DummyMetric2 = new Metric({ namespace: "Dummy", metricName: "Metric2" });
const DummyAxis1: YAxisProps = { min: 0, max: 1 };
const DummyAxis2: YAxisProps = { min: 100, max: 200 };
const DummyAnnotation1: HorizontalAnnotation = { value: 42 };
const DummyAnnotation2: HorizontalAnnotation = { value: 66 };
const DummyVerticalAnnotation: VerticalAnnotation = {
  date: "2021-07-29T02:31:09.890Z",
  color: "123456",
  fill: VerticalShading.BEFORE,
  label: "Custom vertical annotation",
};

test("single axis - single metric: snapshot test", () => {
  const widget = new SingleAxisGraphWidget({
    title: "single axis - single metric",
    width: 4,
    height: 4,
    leftMetrics: [DummyMetric1],
    leftAnnotations: [DummyAnnotation1],
    leftAxis: DummyAxis1,
  });

  expect(removeRegionToken(widget.toJson())).toMatchSnapshot();
});

test("single axis - two metrics: snapshot test", () => {
  const widget = new SingleAxisGraphWidget({
    title: "single axis - two metrics",
    width: 4,
    height: 4,
    leftMetrics: [DummyMetric1, DummyMetric2],
    leftAnnotations: [DummyAnnotation1],
    leftAxis: DummyAxis1,
    verticalAnnotations: [DummyVerticalAnnotation],
  });

  expect(removeRegionToken(widget.toJson())).toMatchSnapshot();
});

test("double axis - two metrics: snapshot test", () => {
  const widget = new DoubleAxisGraphWidget({
    title: "double axis - two metrics",
    width: 4,
    height: 4,
    leftMetrics: [DummyMetric1],
    leftAxis: DummyAxis1,
    leftAnnotations: [DummyAnnotation1],
    rightMetrics: [DummyMetric1],
    rightAxis: DummyAxis2,
    rightAnnotations: [DummyAnnotation2],
    verticalAnnotations: [DummyVerticalAnnotation],
  });

  expect(removeRegionToken(widget.toJson())).toMatchSnapshot();
});

describe("failure test cases", () => {
  test("single axis - no metrics: error", () => {
    expect(
      () =>
        new SingleAxisGraphWidget({
          width: 4,
          height: 4,
          leftAxis: DummyAxis1,
          leftMetrics: [],
        })
    ).toThrowError();
  });

  test("double axis - no left metrics: error", () => {
    expect(
      () =>
        new DoubleAxisGraphWidget({
          width: 4,
          height: 4,
          leftAxis: DummyAxis1,
          leftMetrics: [],
          rightAxis: DummyAxis2,
          rightMetrics: [DummyMetric1],
        })
    ).toThrowError();
  });

  test("double axis - no right metrics: error", () => {
    expect(
      () =>
        new DoubleAxisGraphWidget({
          width: 4,
          height: 4,
          leftAxis: DummyAxis1,
          leftMetrics: [DummyMetric1],
          rightAxis: DummyAxis2,
          rightMetrics: [],
        })
    ).toThrowError();
  });
});

function removeRegionToken(json: any[]): any[] {
  // removing region from the result to avoid TOKEN instability
  (json[0] as any).properties.region = undefined;
  return json;
}
