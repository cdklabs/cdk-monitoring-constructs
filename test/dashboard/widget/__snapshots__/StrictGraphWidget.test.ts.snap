// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`double axis - two metrics: snapshot test 1`] = `
Array [
  Object {
    "height": 4,
    "properties": Object {
      "annotations": Object {
        "horizontal": Array [
          Object {
            "value": 42,
            "yAxis": "left",
          },
          Object {
            "value": 66,
            "yAxis": "right",
          },
        ],
        "vertical": Array [
          Object {
            "color": "123456",
            "fill": "before",
            "label": "Custom vertical annotation",
            "value": "2021-07-29T02:31:09.890Z",
          },
        ],
      },
      "end": undefined,
      "legend": undefined,
      "liveData": undefined,
      "metrics": Array [
        DropEmptyObjectAtTheEndOfAnArray {
          "creationStack": Array [
            "stack traces disabled",
          ],
          "value": Array [
            "Dummy",
            "Metric1",
          ],
        },
      ],
      "period": undefined,
      "region": undefined,
      "setPeriodToTimeRange": undefined,
      "stacked": undefined,
      "start": undefined,
      "stat": undefined,
      "title": "double axis - two metrics",
      "view": "timeSeries",
      "yAxis": Object {
        "left": Object {
          "max": 1,
          "min": 0,
        },
        "right": Object {
          "max": 200,
          "min": 100,
        },
      },
    },
    "type": "metric",
    "width": 4,
    "x": undefined,
    "y": undefined,
  },
]
`;

exports[`single axis - single metric: snapshot test 1`] = `
Array [
  Object {
    "height": 4,
    "properties": Object {
      "annotations": Object {
        "horizontal": Array [
          Object {
            "value": 42,
            "yAxis": "left",
          },
        ],
        "vertical": undefined,
      },
      "end": undefined,
      "legend": Object {
        "position": "hidden",
      },
      "liveData": undefined,
      "metrics": Array [
        DropEmptyObjectAtTheEndOfAnArray {
          "creationStack": Array [
            "stack traces disabled",
          ],
          "value": Array [
            "Dummy",
            "Metric1",
          ],
        },
      ],
      "period": undefined,
      "region": undefined,
      "setPeriodToTimeRange": undefined,
      "stacked": undefined,
      "start": undefined,
      "stat": undefined,
      "title": "single axis - single metric",
      "view": "timeSeries",
      "yAxis": Object {
        "left": Object {
          "max": 1,
          "min": 0,
        },
        "right": undefined,
      },
    },
    "type": "metric",
    "width": 4,
    "x": undefined,
    "y": undefined,
  },
]
`;

exports[`single axis - two metrics: snapshot test 1`] = `
Array [
  Object {
    "height": 4,
    "properties": Object {
      "annotations": Object {
        "horizontal": Array [
          Object {
            "value": 42,
            "yAxis": "left",
          },
        ],
        "vertical": Array [
          Object {
            "color": "123456",
            "fill": "before",
            "label": "Custom vertical annotation",
            "value": "2021-07-29T02:31:09.890Z",
          },
        ],
      },
      "end": undefined,
      "legend": undefined,
      "liveData": undefined,
      "metrics": Array [
        DropEmptyObjectAtTheEndOfAnArray {
          "creationStack": Array [
            "stack traces disabled",
          ],
          "value": Array [
            "Dummy",
            "Metric1",
          ],
        },
        DropEmptyObjectAtTheEndOfAnArray {
          "creationStack": Array [
            "stack traces disabled",
          ],
          "value": Array [
            "Dummy",
            "Metric2",
          ],
        },
      ],
      "period": undefined,
      "region": undefined,
      "setPeriodToTimeRange": undefined,
      "stacked": undefined,
      "start": undefined,
      "stat": undefined,
      "title": "single axis - two metrics",
      "view": "timeSeries",
      "yAxis": Object {
        "left": Object {
          "max": 1,
          "min": 0,
        },
        "right": undefined,
      },
    },
    "type": "metric",
    "width": 4,
    "x": undefined,
    "y": undefined,
  },
]
`;
